File: LeetCode/ReplaceWithGreatestElement/3259.cs
using System;
namespace ReplaceWithGreEle{
    class Program{
    public int[] ReplaceElements(int[] arr) {
       int len = arr.Length; 
       int maxValRight = arr[len-1];
       arr[len-1] = -1;
       
        for(int i= len-2; i>=0; i--){
            int temp = maxValRight;
            if(arr[i]>maxValRight){
                maxValRight = arr[i];
            }
            arr[i] = temp;
        }
        return arr;
    }
        static void Main(){

        }
    }
}

File: LeetCode/ReplaceWithGreatestElement/3259.ts
function replaceElements(arr: number[]): number[] {
  let len = arr.length - 1;
  let greatest = arr[len];
  arr[len] = -1;
  for (let i = len - 1; i >= 0; i--) {
    let temp = greatest;
    if (arr[i] > greatest) {
      greatest = arr[i];
    }
    arr[i] = temp;
  }

  return arr;
}
